activerecord/lib/active_record/relation/batches.rb

```diff
@@ -209,7 +209,7 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
 
       relation = relation.reorder(batch_order).limit(batch_limit)
       relation = apply_limits(relation, start, finish)
-      batch_relation = relation
+      batch_relation = relation.uncached!
```

Comment:
Mutation at the same time we copy it to a second variable seems confusing. Maybe move this call up to 210?

Actually, let's move it to its own line, so we can add a comment: `# Retaining the results in the query cache would undermine the point of batching`

I think that's necessary partly because the tests check it's uncached, but don't really illustrate the *reason* it is. And a comment seems easier than finding a way to make an assertion about query cache bloat. :smile: